concept/type/impl/ThingTypeImpl.java

```diff
@@ -101,7 +103,7 @@ public static ThingTypeImpl of(GraphManager graphMgr, TypeVertex vertex) {
     public void setAbstract() {
         if (!isAbstract()) {
             validateIsNotDeleted();
-            if (getInstances().first().isPresent()) {
+            if (getInstancesExplicit().first().isPresent()) {
```

Comment:
The main change - use `getInstancesExplicit()` instead of `getInstances()` to allow setting abstractness even when a child has instances but the type does not

Reply:
same as above, perhaps flip the `if` and return to break out?

Reply:
I actually prefer the nested style because you can visually "see" the logical dependencies/where the code flows, but happy to change it